dbms/src/Core/Block.cpp

```diff
@@ -541,11 +541,9 @@ Block hstackBlocks(Blocks && blocks, const Block & header)
         return {};
 
     Block res = header.cloneEmpty();
-    size_t num_rows = blocks.front().rows();
```

Comment:
Suggest to add a `res.checkNumberOfRows` before returning the res

Reply:
done